components/post_view/post_header/post_header_custom_status.tsx

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import {FormattedMessage} from 'react-intl';
+
+import {setStatusDropdown} from 'actions/views/status_dropdown';
+import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
+import {makeGetCustomStatus, showPostHeaderUpdateStatusButton, isCustomStatusEnabled} from 'selectors/views/custom_status';
+import {GlobalState} from 'types/store';
+import EmojiIcon from 'components/widgets/icons/emoji_icon';
+
+interface ComponentProps {
+    userId: string;
+    isSystemMessage: boolean;
+}
+
+const PostHeaderCustomStatus = (props: ComponentProps) => {
+    const {userId, isSystemMessage} = props;
+    const getCustomStatus = makeGetCustomStatus();
+    const dispatch = useDispatch();
+    const userCustomStatus = useSelector((state: GlobalState) => getCustomStatus(state, userId));
+    const showUpdateStatusButton = useSelector(showPostHeaderUpdateStatusButton);
+    const customStatusEnabled = useSelector(isCustomStatusEnabled);
+
+    const isCustomStatusSet = userCustomStatus && userCustomStatus.emoji;
+    if (!customStatusEnabled || isSystemMessage) {
+        return null;
+    }
+
+    if (isCustomStatusSet) {
+        return (
+            <CustomStatusEmoji
+                userID={userId}
+                showTooltip={true}
+                emojiSize={14}
+                emojiStyle={{
+                    margin: '4px 0 0 4px',
+                }}
+            />
+        );
+    }
+
+    if (!showUpdateStatusButton) {
+        return null;
+    }
```

Comment:
this can be added to the if-clause above (lines 29 - 31)

Reply:
@michelengelen This must be checked after checking for `isCustomStatusSet` and cannot be merged with the previous if condition in the way that you recommend.

I'll add a comment over this, so no one tries to make that change.